pkg/apis/core/validation/shoot.go

```diff
@@ -1938,14 +1936,67 @@ func validateShootOperationContext(operation string, shoot *core.Shoot, fldPath
 	return allErrs
 }
 
-func validateShootHAControlPlaneUpdate(newMeta, oldMeta metav1.ObjectMeta, fldPath *field.Path) field.ErrorList {
+// ValidateShootHAConfig enforces that both annotation and HA spec are not set together
+func ValidateShootHAConfig(shoot *core.Shoot) field.ErrorList {
```

Comment:
In this function we should also check if a valid value is set for `shoot.Spec.ControlPlane.HighAvailability.FailureTolerance.Type`, similar to how it's done for CRI [here](https://github.com/gardener/gardener/blob/c628f5cf013a1a3242e170872908ace39062fe34/pkg/apis/core/validation/shoot.go#L1699).

Reply:
Since we still support alpha annotation, we will have to validate both (which ever has been set)

Reply:
done